src/cc/libclient/Writer.cc

```diff
@@ -1526,10 +1527,11 @@ class Writer::Impl :
                     return;
                 }
                 if (theStatus == kErrorReadOnly && mClosingFlag &&
-                        0 < mCloseOp.chunkId && mCloseOp.chunkVersion < 0) {
+                        0 < mCloseOp.chunkId && mKeepLeaseFlag) {
```

Comment:
On line 684 you replace `0 < mCloseOp.chunkId && mCloseOp.chunkVersion < 0` with `mKeepLeaseFlag` and here you replace only `mCloseOp.chunkVersion < 0` with `mKeepLeaseFlag`. Shouldn't they be the same?